### ISSUE TITLE
Let DS deploy cluster with < 3 MONs if DEV_ENV is true

### ISSUE THREAD

**Issue by smithfarm** [SUSE/DeepSea#310]
I tried to use `basic-health-ok.sh` (which sets DEV_ENV to the string 'true') to deploy a cluster on a single node, but it failed:

```
2017-06-07T20:53:43.177 INFO:teuthology.orchestra.run.target149202191063.stderr:+ echo '*********************************************'
2017-06-07T20:53:43.178 INFO:teuthology.orchestra.run.target149202191063.stderr:+ echo '********** Running DeepSea Stage 3 **********'
2017-06-07T20:53:43.178 INFO:teuthology.orchestra.run.target149202191063.stderr:+ echo '*********************************************'
2017-06-07T20:53:43.178 INFO:teuthology.orchestra.run.target149202191063.stderr:+ echo ''
2017-06-07T20:53:43.179 INFO:teuthology.orchestra.run.target149202191063.stderr:+ salt-run --no-color state.orch ceph.stage.3
2017-06-07T20:53:43.179 INFO:teuthology.orchestra.run.target149202191063.stderr:+ tee /tmp/stage.3.log
2017-06-07T20:53:43.747 INFO:teuthology.orchestra.run.target149202191063.stderr:[WARNING ] Unable to find IPv6 record for "target149202191063.teuthology" causing a 10 second timeout when rendering grains. Set the dns or /etc/hosts for IPv6 to clear this.
2017-06-07T20:53:44.241 INFO:teuthology.orchestra.run.target149202191063.stderr:[WARNING ] Unable to find IPv6 record for "target149202191063.teuthology" causing a 10 second timeout when rendering grains. Set the dns or /etc/hosts for IPv6 to clear this.
2017-06-07T20:53:44.720 INFO:teuthology.orchestra.run.target149202191063.stderr:[WARNING ] Unable to find IPv6 record for "target149202191063.teuthology" causing a 10 second timeout when rendering grains. Set the dns or /etc/hosts for IPv6 to clear this.
2017-06-07T20:53:45.405 INFO:teuthology.orchestra.run.target149202191063.stderr:[WARNING ] Unable to find IPv6 record for "target149202191063.teuthology" causing a 10 second timeout when rendering grains. Set the dns or /etc/hosts for IPv6 to clear this.
2017-06-07T20:53:46.227 INFO:teuthology.orchestra.run.target149202191063.stderr:[ERROR   ] No highstate or sls specified, no execution made
2017-06-07T20:53:46.237 INFO:teuthology.orchestra.run.target149202191063.stdout:firewall                 : [1m[92mdisabled[0m
2017-06-07T20:53:46.237 INFO:teuthology.orchestra.run.target149202191063.stdout:DEV_ENV                  : [1m[92mTrue[0m
2017-06-07T20:53:46.238 INFO:teuthology.orchestra.run.target149202191063.stdout:fsid                     : [1m[92mvalid[0m
2017-06-07T20:53:46.238 INFO:teuthology.orchestra.run.target149202191063.stdout:public_network           : [1m[92mvalid[0m
2017-06-07T20:53:46.238 INFO:teuthology.orchestra.run.target149202191063.stdout:cluster_network          : [1m[92mvalid[0m
2017-06-07T20:53:46.238 INFO:teuthology.orchestra.run.target149202191063.stdout:cluster_interface        : [1m[92mvalid[0m
2017-06-07T20:53:46.238 INFO:teuthology.orchestra.run.target149202191063.stdout:storage                  : [1m[92mvalid[0m
2017-06-07T20:53:46.238 INFO:teuthology.orchestra.run.target149202191063.stdout:ganesha                  : [1m[92mvalid[0m
2017-06-07T20:53:46.239 INFO:teuthology.orchestra.run.target149202191063.stdout:master_role              : [1m[92mvalid[0m
2017-06-07T20:53:46.239 INFO:teuthology.orchestra.run.target149202191063.stdout:time_server              : [1m[92mvalid[0m
2017-06-07T20:53:46.239 INFO:teuthology.orchestra.run.target149202191063.stdout:fqdn                     : [1m[92mvalid[0m
2017-06-07T20:53:46.239 INFO:teuthology.orchestra.run.target149202191063.stdout:monitors                 : [1m[91m['Too few monitors target149202191086.teuthology'][0m
2017-06-07T20:53:46.239 INFO:teuthology.orchestra.run.target149202191063.stdout:mon_host                 : [1m[91m['Must have at least three entries'][0m
2017-06-07T20:53:46.239 INFO:teuthology.orchestra.run.target149202191063.stdout:mon_initial_members      : [1m[91m['Must have at least three entries'][0m
2017-06-07T20:53:46.240 INFO:teuthology.orchestra.run.target149202191063.stdout:target149202191063.teuthology_master:
2017-06-07T20:53:46.240 INFO:teuthology.orchestra.run.target149202191063.stdout:----------
2017-06-07T20:53:46.240 INFO:teuthology.orchestra.run.target149202191063.stdout:          ID: validate failed
2017-06-07T20:53:46.240 INFO:teuthology.orchestra.run.target149202191063.stdout:    Function: salt.state
2017-06-07T20:53:46.241 INFO:teuthology.orchestra.run.target149202191063.stdout:        Name: just.exit
2017-06-07T20:53:46.241 INFO:teuthology.orchestra.run.target149202191063.stdout:      Result: False
2017-06-07T20:53:46.241 INFO:teuthology.orchestra.run.target149202191063.stdout:     Comment: No highstate or sls specified, no execution made
2017-06-07T20:53:46.241 INFO:teuthology.orchestra.run.target149202191063.stdout:     Started: 20:53:46.228570
2017-06-07T20:53:46.242 INFO:teuthology.orchestra.run.target149202191063.stdout:    Duration: 1.302 ms
2017-06-07T20:53:46.242 INFO:teuthology.orchestra.run.target149202191063.stdout:     Changes:
2017-06-07T20:53:46.242 INFO:teuthology.orchestra.run.target149202191063.stdout:
2017-06-07T20:53:46.243 INFO:teuthology.orchestra.run.target149202191063.stdout:Summary for target149202191063.teuthology_master
2017-06-07T20:53:46.243 INFO:teuthology.orchestra.run.target149202191063.stdout:------------
2017-06-07T20:53:46.243 INFO:teuthology.orchestra.run.target149202191063.stdout:Succeeded: 0
2017-06-07T20:53:46.243 INFO:teuthology.orchestra.run.target149202191063.stdout:Failed:    1
2017-06-07T20:53:46.244 INFO:teuthology.orchestra.run.target149202191063.stdout:------------
2017-06-07T20:53:46.244 INFO:teuthology.orchestra.run.target149202191063.stdout:Total states run:     1
2017-06-07T20:53:46.244 INFO:teuthology.orchestra.run.target149202191063.stdout:Total run time:   1.302 ms
```

**Comment by smithfarm** [comment]
Note: this is blocking work on a DS-based replacement for the `ceph` task.

**Comment by rjfd** [comment]
@smithfarm this has been fixed in PR #307

**Comment by smithfarm** [comment]
@rjfd I was running an old version of DS.

**Comment by rjfd** [comment]
Closing this as it was fixed in #307
